### DISCUSSION TITLE
Disabling Motion and Performing Detection on a Time-Basis?

### DISCUSSION THREAD

**Discussion by TSanders1722** [blakeblackshear/frigate#8158] [Ask A Question] [answered]
Kind of an odd question, but is there a way that Frigate could be used with motion detection disabled at certain times or altogether and instead trigger at say 10Hz and perform detection based on these time-based triggers? I know it's against what Frigate is designed for, but I liked how Frigate's UI is setup and the FFMPEG and go2RTC for multiple cameras and streams is integrated within the build. 
I suppose an application for this could be say performing detection of objects with a camera mounted on a spinning base. Every time the camera completed a rotation, or every x seconds, the camera processes a sequence of frames and performs detection. It is still motion detection in the sense that an object would be tracked in different places, but with the camera not having a steady background to check for object motion against, I wonder if there could be a work around for this.

**Comment by NickM-27** [comment, marked as answer]
Frigate does not send the whole frame to object detection, it would not work well at all to do this.

Also, frigate 0.13 supports auto tracking objects on supported ptz cameras so if that's the goal then frigate can handle the tracking of objects as it controls the camera.

**Reply by TSanders1722** [reply]
I suppose I misunderstood what it sends to object detection then. I know it can have masked areas that it does not care about for object detection. Could you point me to some information on how it sends the frame to detection or a quick overview/place in the repo to review?

**Reply by NickM-27** [reply]
Frigate detects motion and then sends regions (portions of the camera frame) where the motion occurred to object detection. 

This can be seen in three live debug view in frigate where the red boxes are motion boxes and the green boxes are regions

**Reply by TSanders1722** [reply]
Thanks! That makes sense. Are these regions based on the contour area calculations in the improved_motion.py module?

**Reply by NickM-27** [reply]
No, that simply defines what motion box size needs to be to be consisted valid motion. Regions logic is handled in video.py

**Reply by TSanders1722** [reply]
Awesome! Thank you for your input and increasing my understanding. I will mark this as closed.
